tests/snippets/strings.py

```diff
@@ -293,3 +293,12 @@ def try_mutate_str():
 assert next(str_iter_reversed) == "1"
 assert next(str_iter_reversed, None) == None
 assert_raises(StopIteration, lambda: next(str_iter_reversed))
+
+assert str.__rmod__('%i', 30) == "NotImplemented"
+
+typeError = False
+try:
+    str.__rmod__(30, '%i')
+except TypeError:
+    typeError = True
+assert typeError
```

Comment:
You can use `assertRaises` from `testutils` for this:

```python
from testutils import assertRaises

...

with assertRaises(TypeError):
    ...
```

Reply:
Thank you for sharing. I will use your proposal for this PR.